client/my-sites/stats/stats-chart-tabs/index.jsx

```diff
@@ -219,49 +206,42 @@ class StatModuleChartTabs extends Component {
 		} );
 	}
 
+	renderQueries() {
+		const { fullQuery, quickQuery, siteId } = this.props;
+		return (
+			siteId && (
+				<>
+					<QuerySiteStats statType="statsVisits" siteId={ siteId } query={ quickQuery } />
+					<QuerySiteStats statType="statsVisits" siteId={ siteId } query={ fullQuery } />
+				</>
+			)
+		);
+	}
+
 	render() {
-		const { fullQuery, quickQuery, quickQueryRequesting, fullQueryRequesting, siteId } = this.props;
-		const chartData = this.buildChartData();
-		const activeTab = this.getActiveTab();
-		let availableCharts = [];
-		const data = this.getLoadedData();
-		const activeTabLoading =
-			quickQueryRequesting && fullQueryRequesting && ! ( data && data.length );
-		const classes = [
-			'stats-module',
-			'is-chart-tabs',
-			{
-				'is-loading': activeTabLoading,
-			},
-		];
-		if ( activeTab.legendOptions ) {
-			availableCharts = activeTab.legendOptions;
-		}
+		const { isActiveTabLoading } = this.props;
+		const classes = [ 'stats-module', 'is-chart-tabs', { 'is-loading': isActiveTabLoading } ];
 
 		return (
 			<div>
-				{ siteId && (
-					<QuerySiteStats statType="statsVisits" siteId={ siteId } query={ quickQuery } />
-				) }
-				{ siteId && (
-					<QuerySiteStats statType="statsVisits" siteId={ siteId } query={ fullQuery } />
-				) }
 				<Card className={ classNames( ...classes ) }>
+					{ this.renderQueries() }
```

Comment:
is there a reason to not keep these inline? generally we prefer keeping JSX explicit and in the open - it's somewhat frustrating to have to look up around the file and realize it was two lines of static JSX

also, wouldn't this be a great place to use `httpData` and remote the reducers, selectors, data layer handlers, and action creators, for stats info… 🤔

Reply:
This line is in preparation for network optimization coming in #27433. I'll roll this portion back; might be worth investigating how the data layer and #27433 might work together.